File: day1/day1.clj
(:require '[clojure.string :as str])
(as-> "input.txt" $
  (slurp $)
  (str/split $ #"\n\n")
  (map
   #(->> %
         str/split-lines
         (map read-string)
         (reduce +)) $)
  (sort > $)
  (let [x $]
    (println "Part 1:" (first x))
    (println "Part 2:" (reduce + (take 3 x)))))

File: day1/day1.rs
fn main() {
    let mut l: Vec<i32> = std::fs::read_to_string("input.txt")
        .expect("couldn't read file")
        .trim()
        .split("\n\n")
        .map(|x| {
            x.split("\n")
                .map(|y| y.parse::<i32>().expect("couldn't parse int"))
                .sum::<i32>()
        })
        .collect();
    // giga brain stuff from the rust discord, still no non mutating sort :(
    l.select_nth_unstable_by(2, |a, b| b.cmp(a));
    println!("Part 1: {}", &l.first().expect("no first"));
    println!("Part 2: {}", &l.iter().take(3).sum::<i32>())
}
